airbyte-webapp/src/components/connectorBuilder/manifestToBuilderForm.test.ts

```diff
@@ -0,0 +1,434 @@
+import { load } from "js-yaml";
+import merge from "lodash/merge";
+
+import { ConnectorManifest, DeclarativeStream } from "core/request/ConnectorManifest";
+
+import { convertToBuilderFormValues } from "./manifestToBuilderForm";
+import { DEFAULT_BUILDER_FORM_VALUES } from "./types";
+
+const baseManifest: ConnectorManifest = {
+  type: "DeclarativeSource",
+  version: "0.1.0",
+  check: {
+    type: "CheckStream",
+    stream_names: [],
+  },
+  streams: [],
+};
+
+const stream1: DeclarativeStream = {
+  type: "DeclarativeStream",
+  name: "stream1",
+  retriever: {
+    type: "SimpleRetriever",
+    name: "stream1",
+    record_selector: {
+      type: "RecordSelector",
+      extractor: {
+        type: "DpathExtractor",
+        field_pointer: [],
+      },
+    },
+    requester: {
+      type: "HttpRequester",
+      name: "stream1",
+      url_base: "https://url.com",
+      path: "/stream-1-path",
+    },
+  },
+};
+
+const stream2: DeclarativeStream = merge({}, stream1, {
+  name: "stream2",
+  retriever: {
+    name: "stream2",
+    requester: {
+      name: "stream2",
+      path: "/stream-2-path",
+    },
+  },
+});
+
+describe("Conversion throws error when", () => {
```

Comment:
I originally started to add tests for every case that throws a ManifestCompatibilityError, but that felt overly excessive and just means that we would be repeating all of those cases in yet another place, so I decided against it.

I think its more important that we test the more complex conversion cases anyway.